### ISSUE TITLE
Remove items from inventory when used via narrative choices

### ISSUE THREAD

**Issue by jerseycheese** [jerseycheese/Narraitor#920]
## Plain Language Summary
When a player uses or loses an item during gameplay, it should be automatically removed from their inventory. This includes both deliberate usage (choosing "Use Health Potion") and narrative events (AI describes item being stolen, dropped, or consumed).

## Current Behavior
Items remain in inventory even after:
- Player explicitly uses them via choices
- Narrative describes them being consumed, lost, stolen, or destroyed
- Quest items that should disappear after use
- Consumables being used multiple times

This leads to:
- Consumables (potions, food) that can be used infinitely
- Items remaining after AI narrates they were lost/stolen
- Confusion about inventory state vs narrative reality

### Scenario 1: Player Choice Usage
When a player selects a choice that uses an item:
- **Consumables** (potions, food): Reduce quantity by 1, remove if quantity reaches 0
- **Quest items used up**: Remove from inventory entirely
- **Equipment**: Context-dependent (throw sword = remove, wear armor = keep)
- **Non-consumable items**: Stay in inventory (read book doesn't consume it)

### Scenario 2: Narrative-Driven Loss
When the AI narrative describes item loss:
- **"Your sword is knocked from your hand and falls into the chasm"** → Remove sword
- **"The guards confiscate your lockpicks"** → Remove lockpicks
- **"You drink the mysterious potion"** → Remove/reduce potion
- **"The thief steals your gold coins"** → Remove coins

### Detection Methods
1. **AI Metadata** (preferred):
   - Add `itemsLost` metadata to AI responses (similar to `itemsAcquired`)
   - AI indicates: `itemsLost: [{ name: "Iron Sword", reason: "dropped" }]`
   - Automatically process removals after segment generation

2. **Choice-Based** (alternative):
   - Parse choice text for item usage keywords
   - Extract item names from selected choices
   - Less reliable but doesn't require AI changes

### Item Categorization
- **Consumables**: Always removed/decremented when used
- **Equipment**: Removed only if narrative indicates (thrown, stolen, destroyed)
- **Quest items**: Removed when narrative indicates consumption/delivery
- **Documents/Books**: Usually kept unless narrative says otherwise

### Integration Points
- `src/state/narrativeStore.ts` - Process `itemsLost` metadata after segment generation
- `src/state/inventoryStore.ts` - Use existing `deleteItem`/`updateItemQuantity` methods
- `src/lib/ai/narrativeGenerator.ts` - AI prompt for item loss detection
- `src/types/narrative.types.ts` - Add `itemsLost` to `NarrativeMetadata`

## Related Code Areas
- `src/state/inventoryStore.ts` - Item removal logic (already has `deleteItem`, `updateItemQuantity`)
- `src/state/narrativeStore.ts` - Segment/decision processing
- `src/lib/ai/narrativeGenerator.ts` - AI prompt instructions
- `src/types/narrative.types.ts` - Metadata type definitions
- `src/lib/narrative/itemAcquisitionProcessor.ts` - Mirror this for item removal

## Acceptance Criteria
- [ ] AI generates `itemsLost` metadata when narrative describes item usage/loss
- [ ] Consumable items decrease quantity when used via choices
- [ ] Items with quantity 0 are automatically removed from inventory
- [ ] Quest items are removed when narrative indicates consumption/delivery
- [ ] Equipment is removed when narrative describes loss (stolen, dropped, destroyed)
- [ ] Item removal happens automatically after AI segment generation
- [ ] Item removal is reflected immediately in character sheet UI
- [ ] Journal entries note when items are consumed or lost
- [ ] Works correctly for both single items and stacked items
- [ ] Handles both player-initiated usage and narrative-driven loss

## Examples

**Choice-based usage:**
```
Choice: "Use the Health Potion"
Result: Health Potion quantity reduced by 1
```

**Narrative-driven loss:**
```
Narrative: "You throw the torch into the oil, causing a massive explosion."
AI metadata: { itemsLost: [{ name: "Torch", quantity: 1 }] }
Result: Torch removed from inventory
```

**Theft scenario:**
```
Narrative: "The pickpocket makes off with your gold coins before you can react."
AI metadata: { itemsLost: [{ name: "Gold Coins", quantity: 50 }] }
Result: Gold Coins quantity reduced by 50
```

## Priority
Medium - Affects gameplay realism and prevents item duplication exploits

**Comment by jerseycheese** [comment]
✅ Closed by PR #1018

The automated item loss system has been successfully implemented and merged! 🎉

## What Was Implemented:

✅ AI generates `itemsLost` metadata when narrative describes item usage/loss
✅ Consumable items decrease quantity when used via choices  
✅ Items with quantity 0 automatically removed from inventory
✅ Quest items removed when narrative indicates consumption/delivery
✅ Equipment removed when narrative describes loss (stolen, dropped, destroyed)
✅ Item removal happens automatically after AI segment generation
✅ Item removal reflected immediately in character sheet UI
✅ Journal entries created for consumed/lost items
✅ Works for both single items and stacked items
✅ Handles both player-initiated usage and narrative-driven loss

## Key Features:

- **Mirrored Architecture**: Uses same proven pattern as item acquisition system
- **Semantic AI Matching**: Handles natural language ("the sword" → "Steel Sword")
- **Item Loss Inference**: Fallback system infers losses from narrative text when AI forgets metadata
- **Inverse Significance**: Quest item loss = critical journal entry (highly visible to player)
- **Graceful Degradation**: Removes what's available if AI requests more than inventory has
- **Production Logging**: Structured logger with full context for debugging

Thanks for the detailed issue specification! 🚀
